### ISSUE TITLE
Drop Python 2

### ISSUE THREAD

**Issue by lafrech** [marshmallow-code/webargs#440]
Let's wait on #420 to be merged first to avoid a painful rebase.

- [x] Drop Python 2
- [x] Make arguments keyword-only where sensible

Do we also want to drop Python 3.5 ? This can be done in a later major release.

**Comment by sloria** [comment]
Wrote up a checklist recently for this: https://gist.github.com/sloria/b4a5fc2281f86a5733af198d609c5ab3

**Comment by sirosen** [comment]
I wanted to chime in with my $0.02: I wouldn't drop py3.5 support yet because it's the py3 version shipped on several distro versions, notably Ubuntu 16.04.
Since so many people are using Ubuntu as their first distro, and there's the WSL stuff on Windows, there's some reason to give the Ubuntu lifecycle "priority" when looking at the landscape of distros.

By way of example, if someone is using webargs but they're having trouble upgrading from Ubuntu 16.04 infrastructure to Ubuntu 18.04, then they wouldn't be able to move to the new webargs without either (A) fixing their platform version issues or (B) installing another python version alongside the system python. (Yes, they should fix their problems, but my experience is that by the time you're even _talking_ about being stuck on an old platform version, you're in trouble.)

The same argument doesn't apply to forcing the py2 -> py3 transition -- I'm 👍 on dropping py2.

**Comment by lafrech** [comment]
Yeah, no pressure to remove py35 soon. Can be done later.

**Comment by sloria** [comment]
Sounds good. I'm fine keeping support for 3.5 for now.

**Comment by lafrech** [comment]
Summoning @hugovk again.

Would you be willing to tackle this?

Just asking. I don't know if you even use webargs.

**Comment by hugovk** [comment]
Happy to help with Python 3 migration!

Please see PR #454 for the first checkbox item: " Drop Python 2".

The second checkbox "Make arguments keyword-only where sensible" can follow in another PR.

**Comment by sirosen** [comment]
With #472 merged, we can call this done!

I just did a check on the repo looking for _anything_ python2-related I could find and it came up blank.
